holoviews/operation/element.py

```diff
@@ -159,7 +159,7 @@ class apply_when(param.ParameterizedFunction):
     def _apply(self, element, x_range, y_range, invert=False):
         selected = element
         if x_range is not None and y_range is not None:
-            selected = element[x_range, y_range]
+            selected = element.select(x_range=x_range, y_range=y_range)
```

Comment:
This doesn't do anything since `select` indexes based on the dimension name. Also the error you're seeing is completely unrelated to this really. It's a probably related to missing handling for rasterizing empty quadmeshes.